app/controllers/application_controller.rb

```diff
@@ -27,6 +27,38 @@ class ApplicationController < ActionController::Base
 
   helper_method :current_person
 
+  helper_method def url_for(options)
+    space = if options[0].is_a?(Space) && options[0].branded_domain.present?
+      options.delete_at(0)
+    elsif [:edit, :new].include?(options[0]) && options[1].is_a?(Space) && options[1].branded_domain.present?
+      options.delete_at(1)
+    elsif options.is_a?(Space) && options.branded_domain.present?
+      options
+    end
+
+    if space
```

Comment:
You could return early `unless space.present?`, I think it would be a tad more readable

Reply:
I think I need to `return super unless space` but ya!